### ISSUE TITLE
New AudioEngine getCurrentTime() precision BUG (win32)

### ISSUE THREAD

**Issue by redminion** [cocos2d/cocos2d-x#9854]
The precision of AudioEngine::getCurrentTime(id) is not ideal.

The win32 implementation updates currentTime when the audio buffers are rotated, which is set to 0.1 seconds.

This is not adequate for testing. The precision doesn't need to be perfect, but I think calling getCurrentTime() every frame (at 60fps) should return a different value.

I have only looked at the win32 and android implementations, android does not have this limitation.

brought up in thread: http://discuss.cocos2d-x.org/t/new-audioengine-getcurrenttime-precision-win32/19269

**Comment by dzarowny** [comment]
Any progress on this issue?  I am now experiencing issues with setCurrentTime() being slightly off --- I change the time but it continues to play a bit before changing,  I am guessing this is because of the 100ms buffer.  I will create another issue for this if I don't get a response here.

**Comment by dumganhar** [comment]
Yep, it's really a problem. Currently, we have 100ms delay for getting the current time. :(

**Comment by JasonBaby87** [comment]
Hi there,

I just thought I'd bump this up since I'm writing a music game, and the precision is pretty important for it.

**Comment by Xrysnow** [comment]
I got more precise time by using AL_SEC_OFFSET. If anyone is interested, just look at [here](https://github.com/Xrysnow/lstgx_Audio/blob/b5f6f6cf72cc86668282188e2dc40a409adda2c8/AudioPlayer.cpp). It gives a multiple of 20ms on my computer (seems like internal buffer length in OpenAL). There is also a function that can set audio time more precise.
